frameworks/helloworld/tests/test_resource_refinement.py

```diff
@@ -26,10 +38,105 @@ def configure_package(configure_security):
 def test_install():
     config.check_running(config.SERVICE_NAME)
+
+
+@pytest.mark.sanity
+@pytest.mark.smoke
+@pytest.mark.dcos_min_version('1.10')
+def test_marathon_volume_collission():
+    # This test validates that a service registered in a sub-role of
+    # slave_public will _not_ unreserve Marathon volumes.
+
+    # Uninstall HW first
+    sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+
+    # Install the marathon app
+    marathon_app_name = "persistent-test"
+    persistent_app = {
+        "id": marathon_app_name,
+        "mem": 128,
+        "user": "nobody",
+        "cmd": "echo 'this is a test' > persistent-volume/test && sleep 10000",
+        "container": {
+            "type": "MESOS",
+            "docker": {
+                "image": "ubuntu",
+                "forcePullImage": False
+            },
+            "volumes": [
+                {
+                    "persistent": {
+                    "type": "root",
+                    "size": 500,
+                    "constraints": []
+                    },
+                    "mode": "RW",
+                    "containerPath": "persistent-volume"
+                }
+            ]
+        }
+    }
+    try:
+        sdk_marathon.install_app(persistent_app)
+
+        # Get its persistent Volume
+        host = sdk_marathon.get_scheduler_host(marathon_app_name)
+        ok, pv_name = shakedown.run_command_on_agent(host, "ls /var/lib/mesos/slave/volumes/roles/slave_public")
+        assert ok
+
+        pv_name = pv_name.strip()
+
+        @retrying.retry(
+            wait_fixed=1000,
+            stop_max_delay=60*1000
+        )
+        def check_content():
+            ok, pv_content = shakedown.run_command_on_agent(host, "cat /var/lib/mesos/slave/volumes/roles/slave_public/{}/test".format(pv_name))
+            assert pv_content.strip() == "this is a test"
+
+        check_content()
+
+        # Scale down the Marathon app
+        app_config = sdk_marathon.get_config(marathon_app_name)
+        app_config['instances'] = 0
+        sdk_marathon.update_app(marathon_app_name, app_config)
+
+        # Install Hello World
+        sdk_install.install(config.PACKAGE_NAME,
+                            config.SERVICE_NAME,
+                            config.DEFAULT_TASK_COUNT,
+                            additional_options=pre_reserved_options)
+
+        # Make sure the persistent volume is still there
+        check_content()
+
+        # Uninstall Hello World
+        sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+
+        # Make sure the persistent volume is still there
+        check_content()
+
+        # Scale back up the marathon app
+        app_config = sdk_marathon.get_config(marathon_app_name)
+        app_config['instances'] = 1
+        sdk_marathon.update_app(marathon_app_name, app_config)
+
+        # Make sure the persistent volume is still there
+        check_content()
+
+    finally:
```

Comment:
Why does this finally section exist at all?  Prepare for following tests?

Reply:
Yep. Make sure the Marathon app goes away even if the test fails
Make sure HW gets put back in case another test is added to the suite after it.